### PR TITLE
4 authentication

### PR DESCRIPTION
**Pull request by Moniarchy**
-set up POST/register and POST/login routes in index.js file
-set up GET/logout route in index.js file
-created login and register views

### CODE REVIEW

**Comment by Moniarchy** [comment]
@jrob8577 we think this is ready for review.

File: app.js [hunk 1]
```diff
@@ -4,26 +4,49 @@ const favicon = require('serve-favicon');
...
 const cookieParser = require('cookie-parser');
 const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+const passport = require('passport');
+const localStrategy = require('passport-local').Strategy;
 
-const routes = require('../routes/index');
-const users = require('../routes/users');
+
+const routes = require('./routes/index');
```
Comment (jrobcodes) [inline app.js]: Can we rename `routes` here?  I always find this initial naming convention from express-generator confusing...

File: app.js [hunk 2]
```diff
@@ -4,26 +4,49 @@ const favicon = require('serve-favicon');
...
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(express.static(path.join(__dirname, 'public')));
+
+
+app.use('/', routes);
+
+// passport config
+const Account = require('./models/account');
+passport.use(new localStrategy(Account.authenticate()));
```
Comment (jrobcodes) [inline app.js]: Can we move the passport setup out into another module? (See floworky for an example)

File: app.js [hunk 3]
```diff
@@ -4,26 +4,49 @@ const favicon = require('serve-favicon');
...
+passport.serializeUser(Account.serializeUser());
+passport.deserializeUser(Account.deserializeUser());
+
+mongoose.connect(db);
 // app.use(express.static(buildpath+'/public'))
```
Comment (jrobcodes) [inline app.js]: Why is this commented out?

File: database/db.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const db = 'mongodb://localhost/lizardboard'
+const url = 'mongodb://127.0.0.1:27017/lizardboard';
+const MongoClient = require('mongodb').MongoClient, assert = require('assert');
```
Comment (jrobcodes) [inline database/db.js]: remove `assert`, it's not being used

File: database/db.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const db = 'mongodb://localhost/lizardboard'
```
Comment (jrobcodes) [inline database/db.js]: rename `db`

File: database/db.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+const url = 'mongodb://127.0.0.1:27017/lizardboard';
```
Comment (jrobcodes) [inline database/db.js]: Why are there two connection strings?

File: routes/index.js [hunk 7]
```diff
@@ -1,9 +1,43 @@
...
 router.get('/', (request, response, next) => {
   response.render('index', { title: 'Lizardboard' });
 });
 
+router.get('/register', (request, response, next) => {
+  response.render('register', { });
+});
+
+router.post('/register', (request, response, next) => {
+  Account.register(new Account({ username : request.body.username }),
```
Comment (jrobcodes) [inline routes/index.js]: I thought we refactored this to make it more readable?

**Review by jrobcodes — CHANGES_REQUESTED** [review]
@shakalee14 @Moniarchy Changes requested.

**Comment by Moniarchy** [comment]
Closing this PR because @shakalee14 and their partner are working on this anew.

**Pull request closed without merging**